09/homework_09.py

```diff
@@ -0,0 +1,72 @@
+# Створити програму, яка буде приймати число і друкувати відповідне число в послідовності Фібоначчі
+# =============================== Напишем программу за допомогою генератора:
+def fibo(n):
+    a = b = 1                           # першим двум елементам присвоюємо значення 1
+    for i in range(2, n):
+        a, b = b, a + b                 # розрахунок елемента послідовності Фібоначчі
+    yield b                             # змінна b - потрібне нам число
+
+n = int(input("Введіть порядковий номер елемента, який більше нуля: "))
+
+if n > 0:                               # перевіримо чи введений номер більше нуля
+    res = fibo(n)
+                                        # викличем функцію next з результатом нашого генератора - надрукуємо результат
+    print(f"Число Фібоначчі з таким порядковим номером дорівнює: {next(res)}")
+else:
+    print("Порядковий номер не задовольняє умову")
+
+
+# =============================== Напишем программу за допомогою ітератора:
+class Fibo:
+    def __init__(self):
+        self.a = 1
+        self.b = 1
+
+    def __iter__(self):
+        return self
+                                                    # розрахунок числа ФІбоначчі
+    def __next__(self):
+        self.c = self.a
```

Comment:
You can avoid of using additional variables and override a and b inline, like you did with generator:
`self.a, self.b = self.b, self.a + self.b`